xarray/core/dataset.py

```diff
@@ -1933,6 +1933,69 @@ def from_dataframe(cls, dataframe):
             obj[name] = (dims, data)
         return obj
 
+    def to_dict(self):
+        """
+        Convert this dataset to a dictionary following xarray naming conventions.
+        """
+        d = {'coords': {}, 'attrs': dict(self.attrs), 'dims': dict(self.dims), 
+             'data_vars': {}}
+        def time_check(val):
+            if np.issubdtype(val.dtype, np.datetime64):
+                val = val.astype('datetime64[us]')
+            elif np.issubdtype(val.dtype, np.timedelta64):
+                val = val.astype('timedelta64[us]')
+            return(val)
+
+        for k in self.coords:
+            d['coords'].update({k: {'data': time_check(self[k].values).tolist(),
+                                    'dims': self[k].dims,
+                                    'attrs': dict(self[k].attrs)}})
+        for k in self.data_vars:
+            d['data_vars'].update({k: {'data': time_check(self[k].values).tolist(),
+                                       'dims': self[k].dims,
+                                       'attrs': dict(self[k].attrs)}})
+        return d
+
+    @classmethod
+    def from_dict(cls, d):
+        """
+        Convert a dictionary into an xarray.Dataset
+
+        Parameters
+        ----------
+        d : dict, with a minimum structure of {'var_0': {'dims': [..], 'data': [..]}, ..}
+
+        Returns
+        -------
+        obj : xarray.Dataset
+
+        """
+        import itertools
+
+        try:
+            vars = itertools.chain(d['coords'].items(), d['data_vars'].items())
+        except:
```

Comment:
Avoid blanket `except` clauses -- they can catch too many things going wrong. Instead, I would try using the `.get` argument, e.g., `d.get('coords', {})`